pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -337,6 +339,38 @@ func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneLi
 	return nil
 }
 
+func getCPUCoreAttribute(coreKind string) (error, v1alpha2.AttributeInfo) {
+	cpu, err := os.ReadFile(hostpath.SysfsDir.Path("sys/devices/" + coreKind, "cpus"))
+	if os.IsNotExist(err) {
+		klog.ErrorS(err, "error reading cpu entry file", "coreKind", coreKind)
+		return err, v1alpha2.AttributeInfo{}
+	}
+
+	attr := v1alpha2.AttributeInfo{
+		Name:  coreKind,
+		Value: strings.TrimSpace(string(cpu)),
+	}
+
+	return nil, attr
+}
+
+// Dicsover E/P cores
+func discoverCpuCores() v1alpha2.AttributeList {
+	attrList := v1alpha2.AttributeList{}
+
+	err, attrAtom := getCPUCoreAttribute("cpu_atom")
+	if err == nil {
+		attrList = append(attrList, attrAtom)
+	}
```

Comment:
It would feel better to do the error handling here:
```suggestion
	if err, attrAtom := getCPUCoreAttribute("cpu_atom"); err != nil {
		if !os.IsNotExist(err) {
			klog.ErrorS(err, "error reading cpu entry file")
		}
	} else {
		attrList = append(attrList, attrAtom)
	}
```

A possible optimization would be to read all attrs in a loop to save some LOC

```suggestion
for _, name := range []string{"cpu_atom", "cpu_core"} {
	if err, attr := getCPUCoreAttribute("cpu_atom"); err != nil {
		if !os.IsNotExist(err) {
			klog.ErrorS(err, "error reading cpu entry file")
		}
	} else {
		attrList = append(attrList, attr)
	}
}

```

WDYT?

Reply:
there is another optimisation that can be taken: not to hardcode "atom" or "core", but detect what is present on the system and build label based on that. Not critical, but cleaner in my opinion.

Reply:
You mean read all `cpu_*` that are found? Why not. Would eliminate the os.IsNotExist check :)